packages/backend/src/database/migrations/20220428071116_job_steps.ts

```diff
@@ -0,0 +1,46 @@
+import { Knex } from 'knex';
+
+const COMPILE_JOB_STEPS_TABLE_NAME = 'job_steps';
+const COMPILE_JOB_TABLE_NAME = 'jobs';
+
+export async function up(knex: Knex): Promise<void> {
+    // https://github.com/knex/knex/pull/4657
+    await knex.schema.alterTable(COMPILE_JOB_TABLE_NAME, (tableBuilder) => {
+        tableBuilder.uuid('project_uuid').nullable().alter();
+    });
+
+    await knex.schema.createTable(
```

Comment:
there is no constraint here that the jobs can only have one of each `step_type` - would it be needed?

If we can insert the same step twice, then indexing by (`job_uuid`, `step_type`) could return multiple records from job steps

Reply:
![image](https://user-images.githubusercontent.com/1983672/165924331-a04c475a-805e-446c-b186-0c9e8cb795f7.png)